packages/packages/commonwell-sdk/src/models/document.ts

```diff
@@ -0,0 +1,86 @@
+import { z } from "zod";
+import { identifierUseCodesSchema } from "./identifier";
+import { isoDateTimeSchema } from "./iso-datetime";
+import { periodSchema } from "./period";
+
+// Used as a reference, but not exactly the actual definition:
+// https://specification.commonwellalliance.org/services/rest-api-reference#8610-documentreference
+
+// TODO can this be an enum?
+// Bundle, DocumentReference, Organization, Practitioner
+const resourceTypeSchema = z.string().optional();
+
+const docIdentifierSchema = z.object({
+  use: identifierUseCodesSchema.optional(),
+  system: z.string().optional(),
+  value: z.string(),
+});
+
+const containedSchema = z.object({
+  resourceType: resourceTypeSchema,
+  id: z.string(),
+  name: z.string().optional(),
+  organization: z
+    .object({
+      reference: z.string(),
+    })
+    .optional(),
+});
+
+const statusSchema = z.enum(["current", "superceded", "entered in error"]);
+
+// Main Clinical Acts Documented
+const eventSchema = z.object({
+  text: z.string(), // CodeableConcept
+});
+
+export const contentSchema = z.object({
```

Comment:
It looks like we're missing a lot of fields in this model? Why is that?

Also let's keep the names consistent with CW, perhaps `documentReferenceSchema` ?

Reply:
I'm reviewing these... I'm using the actual return from the endpoint as reference and found some discrepancies, so I didn't want to constrain too much, but I think its fine. Reviewing...

Reply:
Reviewed one by one. There's a fair gap between the spec and what we get as a response when querying documents - I'm even thinking that spec is for a document reference, and not the document itself?

This is an example of an actual response:

```json
{
    "resourceType": "Bundle",
    "entry": [
        {
            "id": "urn:uuid:e7f03f5a-db17-43c2-9ad7-d6f941bf64e0",
            "content": {
                "resourceType": "DocumentReference",
                "contained": [
                    {
                        "resourceType": "Organization",
                        "id": "orgRef1",
                        "name": "Metriport-OrgB-1651"
                    },
                    {
                        "resourceType": "Practitioner",
                        "id": "authRef1",
                        "organization": {
                            "reference": "#orgRef1"
                        }
                    }
                ],
                "masterIdentifier": {
                    "system": "urn:ietf:rfc:3986",
                    "value": "2.16.840.1.113883.3.3330.8889429.1651.2^89765..1"
                },
                "identifier": [
                    {
                        "use": "official",
                        "system": "urn:ietf:rfc:3986",
                        "value": "urn:uuid:e7f03f5a-db17-43c2-9ad7-d6f941bf64e0"
                    }
                ],
                "subject": {
                    "reference": "#patRef1"
                },
                "type": {
                    "coding": [
                        {
                            "system": "http://loinc.org/",
                            "code": "34133-9",
                            "display": "Summarization of episode note"
                        }
                    ]
                },
                "author": [
                    {
                        "reference": "#authRef1"
                    }
                ],
                "indexed": "2023-01-31T16:56:50+00:00",
                "status": "current",
                "description": "Metriport-OrgB-1651 Summarization Of Episode Note",
                "mimeType": "application/octet-stream",
                "location": "https://integration.rest.api.commonwellalliance.org/v2/Binary/H4sIAAAAAAAEAJXLOw6AIBBF0d1YvjigOBTGwsoF2NoIiRQwfqBw97oFcsubMx0S_SwxlhTyu7hRgQy4a0Eg0swaf1q3YGbbKfvvnqCa25_yhCz3u6ZwFV9Bnewl-pRr4cZ2MD1AHwauaVW1AAAA0",
                "context": {
                    "event": [
                        {
                            "text": "Problem"
                        },
                        {
                            "text": "drug allergy"
                        },
                        {
                            "text": "Nausea"
                        },
                        {
                            "text": "Depressive disorder"
                        },
                        {
                            "text": "Hypothyroidism"
                        },
                        {
                            "text": "Sinusitis"
                        },
                        {
                            "text": "CT of paranasal sinuses"
                        },
                        {
                            "text": "Left maxillary sinus mass"
                        },
                        {
                            "text": "never a smoker"
                        },
                        {
                            "text": "Vital signs"
                        },
                        {
                            "text": "Neoplasm of accessory sinus"
                        },
                        {
                            "text": "Finding of functional performance and activity"
                        },
                        {
                            "text": "Cane"
                        },
                        {
                            "text": "Cognitive function finding"
                        },
                        {
                            "text": "Short term memory deficit"
                        }
                    ],
                    "period": {
                        "start": "2012-10-02T04:00:00Z"
                    }
                }
            }
        },
        {
            "id": "urn:uuid:b8fa0f9a-9915-40ce-b0de-478d22575cc3",
            "content": {
                "resourceType": "DocumentReference",
                "contained": [
                    {
                        "resourceType": "Organization",
                        "id": "orgRef2",
                        "name": "Metriport-OrgA-1651"
                    },
                    {
                        "resourceType": "Practitioner",
                        "id": "authRef2",
                        "organization": {
                            "reference": "#orgRef2"
                        }
                    }
                ],
                "masterIdentifier": {
                    "system": "urn:ietf:rfc:3986",
                    "value": "2.16.840.1.113883.3.3330.8889429.1651.1^89765..1"
                },
                "identifier": [
                    {
                        "use": "official",
                        "system": "urn:ietf:rfc:3986",
                        "value": "urn:uuid:b8fa0f9a-9915-40ce-b0de-478d22575cc3"
                    }
                ],
                "subject": {
                    "reference": "#patRef2"
                },
                "type": {
                    "coding": [
                        {
                            "system": "http://loinc.org/",
                            "code": "34133-9",
                            "display": "Summarization of episode note"
                        }
                    ]
                },
                "author": [
                    {
                        "reference": "#authRef2"
                    }
                ],
                "indexed": "2023-01-31T16:56:50+00:00",
                "status": "current",
                "description": "Metriport-OrgA-1651 Summarization Of Episode Note",
                "mimeType": "application/octet-stream",
                "location": "https://integration.rest.api.commonwellalliance.org/v2/Binary/H4sIAAAAAAAEAJXLMQ6AIAxA0ds4NlRAy2AcnDyAq4s2kQGqCIO3lyuQP778-ZLAi4RQos_fek494ABkFCAgaiINNa0VEJEzvatsK3WJb3l9lvRt0T-FG9ZTjhI45tZxJzcOFgB_QQwPAbUAAAA1",
                "context": {
                    "event": [
                        {
                            "text": "Problem"
                        },
                        {
                            "text": "drug allergy"
                        },
                        {
                            "text": "Nausea"
                        },
                        {
                            "text": "Depressive disorder"
                        },
                        {
                            "text": "Hypothyroidism"
                        },
                        {
                            "text": "Sinusitis"
                        },
                        {
                            "text": "CT of paranasal sinuses"
                        },
                        {
                            "text": "Left maxillary sinus mass"
                        },
                        {
                            "text": "never a smoker"
                        },
                        {
                            "text": "Vital signs"
                        },
                        {
                            "text": "Neoplasm of accessory sinus"
                        },
                        {
                            "text": "Finding of functional performance and activity"
                        },
                        {
                            "text": "Cane"
                        },
                        {
                            "text": "Cognitive function finding"
                        },
                        {
                            "text": "Short term memory deficit"
                        }
                    ],
                    "period": {
                        "start": "2012-10-02T04:00:00Z"
                    }
                }
            }
        }
    ]
}
```

This is the example from [the docs](https://specification.commonwellalliance.org/services/rest-api-reference#8610-documentreference):

```json
{
    "_links": "link relations",
    "resourceType": "DocumentReference",
    "text": {
        "status": "generated",
        "div": "<div>!-- Snipped for Brevity --></div>"
    },
    "contained": [
        {
            "resourceType": "Practitioner",
            "id": "a1",
            "name": {
                "family": [
                    "Dopplemeyer"
                ],
                "given": [
                    "Sherry"
                ]
            },
            "telecom": [
                {
                    "system": "email",
                    "value": "john.doe@healthcare.example.org"
                }
            ],
            "organization": {
                "display": "Cleveland Clinic"
            },
            "role": [
                {
                    "text": "Primary Surgeon"
                }
            ],
            "specialty": [
                {
                    "text": "Orthopedic"
                }
            ]
        },
        {
            "resourceType": "Practitioner",
            "id": "a2",
            "name": {
                "family": [
                    "Smitty"
                ],
                "given": [
                    "Gerald"
                ]
            },
            "telecom": [
                {
                    "system": "email",
                    "value": "john.doe@healthcare.example.org"
                }
            ],
            "organization": {
                "display": "Cleveland Clinic"
            },
            "role": [
                {
                    "text": "Attending"
                }
            ],
            "specialty": [
                {
                    "text": "Orthopedic"
                }
            ]
        }
    ],
    "masterIdentifier": {
        "system": "urn:ietf:rfc:3986",
        "value": "urn:oid:1.3.6.1.4.1.21367.2005.3.7"
    },
    "subject": {
        "reference": "Patient/xcda"
    },
    "type": {
        "coding": [
            {
                "system": "http://loinc.org",
                "code": "34108-1",
                "display": "Outpatient Note"
            }
        ]
    },
    "author": [
        {
            "reference": "#a1"
        },
        {
            "reference": "#a2"
        }
    ],
    "created": "2005-12-24T09:35:00+11:00",
    "indexed": "2005-12-24T09:43:41+11:00",
    "status": "current",
    "description": "Physical",
    "confidentiality": [
        {
            "coding": [
                {
                    "system": "http://ihe.net/xds/connectathon/confidentialityCodes",
                    "code": "1.3.6.1.4.1.21367.2006.7.101",
                    "display": "Clinical-Staff"
                }
            ]
        }
    ],
    "primaryLanguage": "en-US",
    "mimeType": "application/hl7-v3+xml",
    "size": 3654,
    "hash": "2jmj7l5rSw0yVb/vlWAYkK/YBwk=",
    "location": "http://example.org/xds/mhd/Binary/07a6483f-732b-461e-86b6-edb665c45510",
    "context": {
        "event": [
            {
                "coding": [
                    {
                        "system": "http://ihe.net/xds/connectathon/eventCodes",
                        "code": "T-D8200",
                        "display": "Arm"
                    }
                ]
            }
        ],
        "period": {
            "start": "2004-12-23T08:00:00",
            "end": "2004-12-23T08:01:00"
        },
        "facilityType": {
            "coding": [
                {
                    "system": "http://www.ihe.net/xds/connectathon/healthcareFacilityTypeCodes",
                    "code": "Outpatient",
                    "display": "Outpatient"
                }
            ]
        }
    }
}
```

To analyze these I double click on VSCode's open file tab list (on an empty space), it opens a new tab for a new file; select `json` as language; paste it; auto format.

Reply:
Discussed offline, we'll update this as necessary based on what we see in prod.